Error: warnings has no attribute DeprecationWarning

While testing `pluggy` off the branch from #72, I came across this error:

```
        if '__multicall__' in hookimpl.argnames:
            warnings.warn(
                "Support for __multicall__ is now deprecated and will be"
                "removed in an upcoming release.",
>               warnings.DeprecationWarning
            )
E           AttributeError: module 'warnings' has no attribute 'DeprecationWarning'
```


https://github.com/pytest-dev/pluggy/blob/557cd3f0245347c1cd359c6deb4b25cde944e98a/pluggy/__init__.py#L656-L661

And indeed `warnings` module doesn't have a `DeprecationWarning`.

We should add a test to catch this warnings (and any others we might have).

--- a/pluggy/__init__.py
+++ b/pluggy/__init__.py
@@ -657,7 +657,7 @@ class _HookCaller(object):
             warnings.warn(
                 "Support for __multicall__ is now deprecated and will be"
                 "removed in an upcoming release.",
-                warnings.DeprecationWarning
+                DeprecationWarning
             )
             self.multicall = _LegacyMultiCall
 

--- a/testing/test_pluginmanager.py
+++ b/testing/test_pluginmanager.py
@@ -345,8 +345,7 @@ def test_multicall_deprecated(pm):
             assert not __multicall__.hook_impls
             return 17
 
-    with pytest.deprecated_call():
-        pm.register(P1())
+    pytest.deprecated_call(pm.register, P1())
 
 
 def test_add_hookspecs_nohooks(pm):

--- a/testing/test_pluginmanager.py
+++ b/testing/test_pluginmanager.py
@@ -341,9 +341,7 @@ def test_multicall_deprecated(pm):
     class P1(object):
         @hookimpl
         def m(self, __multicall__, x):
-            assert len(__multicall__.results) == 1
-            assert not __multicall__.hook_impls
-            return 17
+            pass
 
     pytest.deprecated_call(pm.register, P1())
 

--- a/testing/test_pluginmanager.py
+++ b/testing/test_pluginmanager.py
@@ -337,6 +337,18 @@ def test_subset_hook_caller(pm):
     assert l == [10]
 
 
+def test_multicall_deprecated(pm):
+    class P1(object):
+        @hookimpl
+        def m(self, __multicall__, x):
+            assert len(__multicall__.results) == 1
+            assert not __multicall__.hook_impls
+            return 17
+
+    with pytest.deprecated_call():
+        pm.register(P1())
+
+
 def test_add_hookspecs_nohooks(pm):
     with pytest.raises(ValueError):
         pm.add_hookspecs(10)

